hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomCallOperation.java

```diff
@@ -88,8 +88,7 @@ private boolean mustBePresent(@NonNull final MessageFrame frame, @NonNull final
         }
         // Let system accounts calls or if configured to allow calls to non-existing contract address calls
         // go through so the message call processor can fail in a more legible way
-        return !addressChecks.isSystemAccount(toAddress)
-                && proxyUpdaterFor(frame).contractMustBePresent();
+        return !addressChecks.isSystemAccount(toAddress) && contractRequired(frame, toAddress, featureFlags);
```

Comment:
so ... if it _is_ a system account does _not_ need to be present
and then
if it _is_ a "contract required" address then does need to be present

had to write that down to figure out what was going on, and if it matched the comment

I found this rather tricky to read.  In fact I'm still not sure I'm reading it correctly.